lib/appmap/event.rb

```diff
@@ -276,15 +327,18 @@ class MethodReturn < MethodReturnIgnoreValue
       attr_accessor :return_value, :exceptions
 
       class << self
-        def build_from_invocation(parent_id, return_value, exception, elapsed: nil, event: MethodReturn.new)
+        def build_from_invocation(parent_id, return_value, exception, elapsed: nil, event: MethodReturn.new, parameter_schema: false)
           event ||= MethodReturn.new
           event.tap do |_|
             if return_value
               event.return_value = {
                 class: best_class_name(return_value),
                 value: display_string(return_value),
                 object_id: return_value.__id__
-              }
+              }.tap do |param|
+                param[:size] = return_value.size if return_value.respond_to?(:size) && return_value.is_a?(Enumerable)
```

Comment:
This is now handled by `add_schema` so you can remove this line. I asked for this precisely because of this duplication :)

Reply:
This line and schema are not that closely interrelated in the new iteration.